cmd/kots/cli/plugin.go

```diff
@@ -0,0 +1,201 @@
+package cli
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"os"
+
+	"github.com/pkg/errors"
+	"github.com/replicatedhq/kots/pkg/kotsutil"
+	"github.com/replicatedhq/kots/pkg/logger"
+	"github.com/replicatedhq/kots/pkg/preflight"
+	"github.com/replicatedhq/kots/pkg/template"
+	analyzerunner "github.com/replicatedhq/troubleshoot/pkg/analyze"
+	"github.com/replicatedhq/troubleshoot/pkg/oci"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func PluginCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:           "plugin",
+		Short:         "Run commands for helm plugin",
+		Long:          `Run commands for helm plugin, this serves as an entry point for subcommands such as preflight`,
+		SilenceUsage:  false,
+		SilenceErrors: false,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			viper.BindPFlags(cmd.Flags())
+		},
+		RunE: func(cmd *cobra.Command, args []string) error {
+			if len(args) != 1 {
+				cmd.Help()
+				os.Exit(1)
+			}
+			log := logger.NewCLILogger(cmd.OutOrStdout())
+
+			if args[0] == "preflight" {
+				preflightFile, err := cmd.Flags().GetString("preflight")
+				if err != nil {
+					log.Errorf("failed to get preflight flag: %v\n", err)
+					return err
+				}
+
+				configFile, err := cmd.Flags().GetString("config")
+				if err != nil {
+					log.Errorf("failed to get config flag: %v\n", err)
+					return err
+				}
+
+				err = renderAndRunPreflight(preflightFile, configFile)
+				if err != nil {
+					log.Errorf("failed to render and run preflight %s: %v\n", preflightFile, err)
+					return err
+				}
+			} else if args[0] == "config" {
+				// TODO: logic for initial install for pulling config yaml
+			}
+			return nil
+		},
+	}
+
+	cmd.Flags().String("preflight", "", "the filename or url of the Preflight spec")
+	cmd.Flags().String("config", "", "the filename of the Config spec")
+
+	return cmd
+}
+
+func renderAndRunPreflight(preflightFile, configFile string) error {
+	isUrl := true
+	parsed, err := url.ParseRequestURI(preflightFile)
+	if err != nil {
+		isUrl = false
+	}
+
+	var prefBytes []byte
+	if isUrl && parsed != nil && parsed.Scheme == "oci" {
```

Comment:
All of this should be in the troubleshoot repo.  It's already implemented there.  It might be sufficient to export [this function](https://github.com/replicatedhq/troubleshoot/blob/374a65522dda8342a9893d96d6df756f224e4b74/cmd/preflight/cli/run.go#L34). Though it might be better to move it into a separate package and remove cobra references from it.  BUT we could also import [these](https://github.com/replicatedhq/troubleshoot/blob/3a5816cd18ddb991e3681a8d6d76d35e247ce680/cmd/preflight/cli/root.go#L40-L49) flags from troubleshoot.  And make this integration magical just like when we import kubectl flags in our code instead of redefining them.

Reply:
ya im going to move the code in troubleshoot into the preflight package in that repo, export whats necessary, and remove cobra references from the funcs and pass them in instead

Reply:
PR for this: https://github.com/replicatedhq/troubleshoot/pull/769